File: app/src/main/java/com/droidtitan/volleyexamples/rest/di/AppComponent.java
package com.droidtitan.volleyexamples.rest.di;

import com.droidtitan.volleyexamples.rest.fragment.ImageLoaderFragment;
import com.droidtitan.volleyexamples.rest.fragment.NetworkImageFragment;
import com.droidtitan.volleyexamples.rest.fragment.RequestFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = RestModule.class)
public interface AppComponent {
    void inject(ImageLoaderFragment fragment);

    void inject(NetworkImageFragment fragment);

    void inject(RequestFragment fragment);
}

File: app/src/main/java/com/droidtitan/volleyexamples/rest/di/AppComponent.kt
package com.droidtitan.volleyexamples.rest.di

import com.droidtitan.volleyexamples.rest.fragment.GsonRequestFragment
import com.droidtitan.volleyexamples.rest.fragment.ImageLoaderFragment
import com.droidtitan.volleyexamples.rest.fragment.NetworkImageFragment
import dagger.Component
import javax.inject.Singleton

Singleton
Component(modules = arrayOf(RestModule::class))
public interface AppComponent {
    public fun inject(f: ImageLoaderFragment)

    public fun inject(f: NetworkImageFragment)

    public fun inject(f: GsonRequestFragment)
}
